src/react/ui-elements/Veeam/VeeamCapacityModal.test.tsx

```diff
@@ -0,0 +1,64 @@
+import { fireEvent, render, screen, waitFor } from '@testing-library/react';
+import { rest } from 'msw';
+import { setupServer } from 'msw/node';
+import { bucketName } from '../../../js/mock/S3Client';
+import { NewWrapper, TEST_API_BASE_URL } from '../../utils/testUtil';
+import { VeeamCapacityModal } from './ VeeamCapacityModal';
+import { VEEAM_XML_PREFIX } from './VeeamConstants';
+
+describe('VeeamCapacityModal', () => {
+  const mockClose = jest.fn();
+  const mockMutate = jest.fn();
+  const server = setupServer(
+    rest.put(
+      `${TEST_API_BASE_URL}/${bucketName}/${VEEAM_XML_PREFIX}/capacity.xml`,
+      (req, res, ctx) => {
+        mockMutate(req.body);
+        return res(ctx.status(200));
+      },
+    ),
+  );
+  beforeAll(() => {
+    server.listen({ onUnhandledRequest: 'error' });
+  });
+  afterEach(() => {
+    server.resetHandlers();
+  });
+  afterAll(() => server.close());
+
+  const selectors = {
+    modalTitle: () => screen.getByText('Edit max repository capacity'),
+    cancelBtn: () => screen.getByText('Cancel'),
+    editBtn: () => screen.getByText('Edit'),
+  };
+
+  beforeEach(() => {
+    render(
+      <VeeamCapacityModal
+        isOpen={true}
+        close={mockClose}
+        bucketName={bucketName}
+        maxCapacity="100 GiB"
+      />,
+      {
+        wrapper: NewWrapper(),
+      },
+    );
+  });
+
+  it('should render the modal', () => {
+    expect(selectors.modalTitle()).toBeInTheDocument();
+  });
+
+  it('should call close function when cancel button is clicked', () => {
+    fireEvent.click(selectors.cancelBtn());
+    expect(mockClose).toHaveBeenCalled();
+  });
+
+  it('should call mutate function when edit button is clicked', async () => {
+    fireEvent.click(selectors.editBtn());
+    await waitFor(async () => {
+      expect(mockMutate).toHaveBeenCalled();
+    });
+  });
```

Comment:
We should also test what happens when the mutation fails ?
To be consistent porbably we should add a banner in the modal ?

Reply:
Fixed in 49e7d8dae934e9182e815b5ad0f47c51712c374b